launcher/src/main/java/org/apache/spark/launcher/OutputRedirector.java

```diff
@@ -17,6 +17,8 @@
 
 package org.apache.spark.launcher;
 
+import org.apache.commons.lang.StringUtils;
```

Comment:
This module shouldn't really depend on any 3rd-party libraries... there's an oddity in the current parent pom that causes this dependency to be available but I think we should instead fix that by adding this property to launcher/pom.xml:

```
    <hadoop.deps.scope>test</hadoop.deps.scope>
```

`containsIgnoreCase` shouldn't really be hard to replicate. Although, if you really want to simplify things, I'd just set the error in this case to a generic "Spark exited with error code X. Please check logs.".

Reply:
copied the code into a helper method